Api/Services/RestaurantService.cs

```diff
@@ -202,83 +202,88 @@ public async Task<List<RestaurantSummaryVM>> GetMyRestaurantsAsync(User user)
-                };
-            }
+                    return result;
+                }
 
-            var restaurantOwnerId = await context.Restaurants
-                .Where(r => r.Id == restaurantId)
-                .Select(r => r.Group!.OwnerId)
-                .FirstOrDefaultAsync();
-            if (restaurantOwnerId is null)
-            {
-                return new List<ValidationResult>
+                var restaurantOwnerId = await context.Restaurants
+                    .Where(r => r.Id == restaurantId)
+                    .Select(r => r.Group!.OwnerId)
+                    .FirstOrDefaultAsync();
+
+
+                if (restaurantOwnerId is null)
                 {
-                    new($"Restaurant with ID {restaurantId} not found")
-                };
-            }
+                    return new ValidationFailure
+                    {
+                        PropertyName = nameof(restaurantOwnerId),
+                        ErrorCode = ErrorCodes.NotFound
+                    };
+                }
 
-            if (restaurantOwnerId != employerId)
-            {
-                return new List<ValidationResult>
+                if (restaurantOwnerId != employerId)
                 {
-                    new("User is not the owner of the restaurant")
-                };
-            }
+                    return new ValidationFailure
+                    {
+                        PropertyName = nameof(employerId),
+                        ErrorCode = ErrorCodes.AccessDenied
+                    };
+                }
 
-            var employee = await context.Users.FindAsync(request.Id);
-            if (employee is null)
-            {
-                return new List<ValidationResult>
+                var employee = await context.Users.FindAsync(request.Id);
+                if (employee is null)
                 {
-                    new($"User with ID {request.Id} not found",
-                        [nameof(request.Id)])
-                };
-            }
+                    return new ValidationFailure
+                    {
+                        PropertyName = nameof(employee),
+                        ErrorCode = ErrorCodes.NotFound
+                    };
+                }
 
-            if (!await userManager.IsInRoleAsync(employee, Roles.RestaurantEmployee)
-                || employee.EmployerId != employerId)
-            {
-                return new List<ValidationResult>
+                if (!await userManager.IsInRoleAsync(employee, Roles.RestaurantEmployee)
+                    || employee.EmployerId != employerId)
                 {
-                    new($"User with ID {request.Id} is not current user's employee",
-                        [nameof(request.Id)])
-                };
-            }
+                    return new ValidationFailure
+                    {
+                        PropertyName = nameof(employerId),
```

Comment:
tak samo jak i z innymi property name w tym przypadku odpowiednie property to request.Id